pkg/scalers/external_scaler.go

```diff
@@ -298,22 +293,11 @@ var connectionPoolMutex sync.Mutex
 
 // getClientForConnectionPool returns a grpcClient and a done() Func. The done() function must be called once the client is no longer
 // in use to clean up the shared grpc.ClientConn
-func getClientForConnectionPool(metadata externalScalerMetadata, logger logr.Logger) (pb.ExternalScalerClient, error) {
+func getClientForConnectionPool(metadata externalScalerMetadata) (pb.ExternalScalerClient, error) {
 	connectionPoolMutex.Lock()
 	defer connectionPoolMutex.Unlock()
 
 	buildGRPCConnection := func(metadata externalScalerMetadata) (*grpc.ClientConn, error) {
-		// FIXME: DEPRECATED to be removed in v2.13 https://github.com/kedacore/keda/issues/4549
-		if metadata.tlsCertFile != "" {
-			logger.V(1).Info("tlsCertFile in ScaleObject metadata will be deprecated in v2.12. Please use" +
-				"tlsClientCert, tlsClientKey and caCert in TriggerAuthentication instead.")
-			creds, err := credentials.NewClientTLSFromFile(metadata.tlsCertFile, "")
-			if err != nil {
-				return nil, err
-			}
-			return grpc.Dial(metadata.scalerAddress, grpc.WithTransportCredentials(creds))
-		}
```

Comment:
AFAIR, we can't delete this yet. Am I right @zroubalik @tomkerkhove ?

Reply:
Correct

Reply:
Let me revert all the changes in this file `pkg/scalers/external_scaler.go`